beetmoverscript/script.py

```diff
@@ -247,10 +249,32 @@ def list_bucket_objects(context, s3_resource, prefix):
 # move_beets {{{1
 async def move_beets(context, artifacts_to_beetmove, manifest):
     beets = []
+
     for locale in artifacts_to_beetmove:
+
+        installer_path = ''
+
+        # get path of installer beet
+        for artifact in artifacts_to_beetmove[locale]:
+            # import pdb; pdb.set_trace()
+            if artifact in INSTALLER_ARTIFACTS:
+                installer_path = artifacts_to_beetmove[locale][artifact]
+
+        # error check
+        if not installer_path:
```

Comment:
Hm, this is wrong. This `move_beets` function is (re)used by other beetmover jobs (e.g. `beetmover-checksums`, `beetmover-source`, `beetmover-signed-langpacks-nightly`, etc), all of which don't have the installer within their `upstreamArtifacts`. They will barf at this check and fail. I glanced at the last staging releases and I can confirm that happens. See https://taskcluster-artifacts.net/LXvVyIerRmCEfuCWbrUUQw/0/public/logs/live_backing.log (one of the errors logs for `beetmover-checksums). Or just glance at the staging releases "Failed" tasks and there are many.

What you can do here is a) check if buildhub artifact is present in the upstreamArtifacts. b) check if the installer is present. You can then raise exception if buildhub is present but not the installer path.